src/main/java/games/battlelore/actions/AttackUnitsAction.java

```diff
@@ -0,0 +1,120 @@
+package games.battlelore.actions;
+
+import core.AbstractGameState;
+import core.actions.AbstractAction;
+import games.battlelore.BattleloreGameState;
+import games.battlelore.cards.CommandCard;
+import games.battlelore.components.CombatDice;
+import games.battlelore.components.MapTile;
+import games.battlelore.components.Unit;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Random;
+
+public class AttackUnitsAction extends AbstractAction {
+    private Unit.Faction playerFaction;
+    private MapTile attacker;
```

Comment:
This is a component, and functionality will break if passed by reference. Should use component ID instead, and then find it in the game state when executing the action (findComponentByID method on the base game state class)

Reply:
Clarifying this: the class itself (AttackUnitsAction) needs to store primitive types (e.g. attack unit tile ID, rather than the tile itself) - the actual objects can then be retrieved by ID when executing the action.